src/NuGet.Indexing/Extraction/CatalogPackageMetadataExtraction.cs

```diff
@@ -55,18 +57,52 @@ private class Extractor
                 AddString(MetadataConstants.ReleaseNotesPropertyName);
                 AddString(MetadataConstants.CopyrightPropertyName);
                 AddString(MetadataConstants.LanguagePropertyName);
-                AddString(MetadataConstants.LicenseUrlPropertyName);
                 AddString(MetadataConstants.PackageHashPropertyName);
                 AddString(MetadataConstants.PackageHashAlgorithmPropertyName);
                 AddString(MetadataConstants.PackageSizePropertyName);
                 AddString(MetadataConstants.CatalogMetadata.RequiresLicenseAcceptancePropertyName, MetadataConstants.RequiresLicenseAcceptancePropertyName);
 
+                AddLicenseUrl();
                 AddFlattenedDependencies();
                 AddSupportedFrameworks();
 
                 return _metadata;
             }
 
+            private void AddLicenseUrl()
+            {
+                if (_galleryBaseAddress != null && _catalog[MetadataConstants.LicenseExpressionPropertyName] != null)
```

Comment:
This if statement can be simplified:
  if (_galleryBaseAddress != null && 
(_catalog[MetadataConstants.LicenseExpressionPropertyName] != null || _catalog[MetadataConstants.LicenseFilePropertyName] != null))
{
_metadata[MetadataConstants.LicenseUrlPropertyName] = GetOverWriteLicenseUrl();
}
else
{
AddString(MetadataConstants.LicenseUrlPropertyName);
}